syntax = "proto3";

package gugong;

// 朝向
enum Orientation {
    NORTH = 0;
    WEST = 1;
    SOUTH = 2;
    EAST = 3;
    UNDEFINED_ORIENTATION = 100;
}

// 建筑类型
enum ArchType {
    // 门 - Gate
    MEN = 0;
    // 殿
    DIAN = 1;
    // 亭 - Kiosk
    TING = 2;
    // 斋
    ZHAI = 3;
    // 宫 - Palace
    GONG = 4;
    // 阁
    GE = 5;
    // 轩
    XUAN = 6;
    // 楼
    LOU = 7;
    // 堂
    TANG = 8;
    // 馆
    GUAN = 9;
    // 桥 - Bridge
    QIAO = 10;
    // 花园 - Garden
    HUAYUAN = 11;
    // 未定义 - Undefined
    UNDEFINED_ARCH_TYPE = 100;
}

// 区域
enum Section {
    FRONTIER = 0;
    FRONT_THREE_DIAN = 1;
    REAR_THREE_PALACES = 2;
    WENHUA_DIAN = 3;
    WUYING_DIAN = 4;
    EAST_6_PALACIES = 5;
    WEST_6_PALACIES = 6;
    EMPIRIER_GARDEN = 7;
    NINGSHOU_PALACE = 8;
    ZHAI_PALACE = 9;
    YANG_XIN_DIAN = 10;
}

// 朝代
enum enumDynasty {
    MING = 0;
    QING = 1;
}

// 皇帝
enum enumEmpire {
    EP_QING_QIANLONG = 0;
    EP_QING_GUANGXU = 12;
}

// 面积
message Area {
    float depth = 1;
    float width = 2;
    int32 depth_unit = 3;
    int32 width_unit = 4;
}

// 皇帝
message Empire {
    string id = 1;
    // 姓名
    string name = 2;
    // 朝代
    enumDynasty dynasty = 3;
    // 年号
    string epoch_name = 4;
    // 庙号
    string temple_name = 5;
}

// 地点
message Place {
    string id = 1;
    // 名称
    string name = 2;
    // 其它名称
    repeated string other_names = 3;
    // 建筑类型
    ArchType arch_type = 4;
    // 正门朝向
    Orientation orientation = 5;
    // 区域
    Section section = 6;
    Area area = 7;
}

// 宫
message Palace {
    string id = 1;
    string name = 2;
}

// 门
message Gate {
    string id = 1;
    string name = 2;
}

// 亭
message Kiosk {
    string id = 1;
    string name = 2;
}
